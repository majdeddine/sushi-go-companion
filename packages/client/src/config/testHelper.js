import React from "react";
import renderer from "react-test-renderer";
import { ThemeProvider } from "styled-components";
import { Theme } from "../resources/styles/theme";

const renderWithTheme = component =>
  renderer.create(<ThemeProvider theme={Theme}>{component}</ThemeProvider>);

export { renderWithTheme };
