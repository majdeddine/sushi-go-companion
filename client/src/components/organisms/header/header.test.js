import React from 'react'
import { renderWithTheme } from '../../../config/testHelper'
import StyledHeader from './styles'

describe('<StyledHeader/>', () => {
  it('Should render a StyledHeader element with default props', () => {
    const component = renderWithTheme(
      <StyledHeader>
        <span role="img" aria-label="bunny emoji">
          üê∞
        </span>
      </StyledHeader>,
    ).toJSON()
    expect(component).toMatchSnapshot()
  })
})
