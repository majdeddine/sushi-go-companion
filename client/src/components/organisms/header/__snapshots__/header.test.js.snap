// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StyledHeader/> Should render a StyledHeader element with default props 1`] = `
.c0 {
  height: 50px;
  background-color: #c51162;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 16px;
}

.c0 > div:last-of-type:not(:only-child) {
  margin-left: auto;
}

<header
  className="c0"
>
  <span
    aria-label="bunny emoji"
    role="img"
  >
    🐰
  </span>
</header>
`;
